modules/csg/csg.cpp

```diff
@@ -523,7 +523,7 @@ void CSGBrushOperation::MeshMerge::_add_distance(List<real_t> &r_intersectionsA,
 
 	// Check if distance exists.
 	for (const List<real_t>::Element *E = intersections.front(); E; E = E->next()) {
-		if (Math::abs(**E - p_distance) < vertex_snap) {
+		if (Math::abs(**E - p_distance) < CMP_EPSILON) {
```

Comment:
Could be `if (Math::is_equal_approx(**E, p_distance)) {`

Reply:
Updated to use is_equal_approx().